lib/services/platform-service.ts

```diff
@@ -746,27 +746,50 @@ export class PlatformService extends EventEmitter implements IPlatformService {
 		return platformData.platformProjectService.isPlatformPrepared(platformData.projectRoot, projectData);
 	}
 
-	private getApplicationPackages(buildOutputPath: string, validPackageNames: string[]): IApplicationPackage[] {
+	private getApplicationPackages(buildOutputPath: string, validBuildOutputData: IValidBuildOutputData): IApplicationPackage[] {
 		// Get latest package` that is produced from build
-		const candidates = this.$fs.readDirectory(buildOutputPath);
-		const packages = _.filter(candidates, candidate => {
-			return _.includes(validPackageNames, candidate);
-		}).map(currentPackage => {
-			currentPackage = path.join(buildOutputPath, currentPackage);
-
-			return {
-				packageName: currentPackage,
-				time: this.$fs.getFsStats(currentPackage).mtime
-			};
-		});
+		let result = this.getApplicationPackagesCore(this.$fs.readDirectory(buildOutputPath).map(filename => path.join(buildOutputPath, filename)), validBuildOutputData.packageNames);
+		if (result) {
+			return result;
+		}
+
+		const candidates = this.$fs.enumerateFilesInDirectorySync(buildOutputPath);
+		result = this.getApplicationPackagesCore(candidates, validBuildOutputData.packageNames);
+		if (result) {
+			return result;
+		}
+
+		if (validBuildOutputData.regexes && validBuildOutputData.regexes.length) {
+			return this.createApplicationPackages(candidates.filter(filepath => _.some(validBuildOutputData.regexes, regex => regex.test(path.basename(filepath)))));
+		}
+
+		return [];
+	}
 
-		return packages;
+	private getApplicationPackagesCore(candidates: string[], validPackageNames: string[]): IApplicationPackage[] {
+		const packages = candidates.filter(filePath => _.includes(validPackageNames, path.basename(filePath)));
+		if (packages.length > 0) {
+			return this.createApplicationPackages(packages);
+		}
+
+		return null;
+	}
+
+	private createApplicationPackages(packages: string[]): IApplicationPackage[] {
+		return packages.map(filepath => this.createApplicationPackage(filepath));
 	}
 
-	private getLatestApplicationPackage(buildOutputPath: string, validPackageNames: string[]): IApplicationPackage {
-		let packages = this.getApplicationPackages(buildOutputPath, validPackageNames);
+	private createApplicationPackage(filepath: string): IApplicationPackage {
```

Comment:
This method's argument could be named `packageName` in order to enable a shorter syntax down below:
```ts
return {
    packageName,
    time: this.$fs.getFsStats(packageName).mtime
};
```